R/qview.R

```diff
@@ -0,0 +1,214 @@
+globalVariables(names = c("id", "user", "cancel_button"))
+
+## Support Functions ####
+
+#' Function to get the user, or a mock user when the user isn't available.
+#' @param session The shiny session object.
+#' @param .mock_user The "default" username to use if one is not available in session$user.
+#' @export
+whoami <- function(session, .mock_user = "UNKNOWN") {
+  if (!is.null(session$user)) {
+    return(session$user)
+  }
+  return(.mock_user)
+}
+
+#' Function to create a cancel button to be rendered in a Data Table.
+#' @param index The index of the Data Table row that the button will be placed in.
+#' @param .ns The namespace function that should be applied to the button's ID.
+#' @details
+#' The function returns the cancel button as raw HTML with a JavaScript statement
+#' embedded. Using raw HTML + JavaScript allows us to embed the button directly into
+#' a DataTable object.
+#'
+cancel_button_creator <- function(index, .ns) {
+  return(
+    as.character(
+      shiny::actionButton(
+        glue::glue("cancel_button_{as.character(index)}"),
+        glue::glue("Cancel"),
+        row_index = index,
+        onclick = paste0('Shiny.onInputChange(\"', .ns("cancel_button"), '\",  this.attributes.row_index.value)')
+      )
+    )
+  )
+}
+
+#' Function to grab a dataframe of the jobs in the queue for the given user.
+#' @param queue_obj The RSimpleQueue object currently in use.
+#' @param .user THe user as a character string -- only this user's jobs will be returned.
+#' @importFrom dplyr filter arrange desc
+#' @importFrom purrr is_null
+get_queue_data <- function(queue_obj, .user) {
```

Comment:
This would be more "testable" if I passed in a list of jobs instead of the queue_obj that is used to _get_ a list of jobs. I managed to mock my way around this, but what do you think @dpastoor ?

Reply:
I think I prefer it the way it is because it's simpler from the top-level (the "server" code), but objectively speaking it's probably better to have the function be a pure function. Unless I'm told otherwise, I'm going to leave it like this for now and revisit it later.